packages/block-library/src/cover/edit.js

```diff
@@ -164,18 +199,12 @@ class CoverEdit extends Component {
 						<>
 							<MediaUploadCheck>
 								<Toolbar>
-									<MediaUpload
-										onSelect={ onSelectMedia }
-										allowedTypes={ ALLOWED_MEDIA_TYPES }
-										value={ id }
-										render={ ( { open } ) => (
-											<IconButton
-												className="components-toolbar__control"
-												label={ __( 'Edit media' ) }
-												icon="edit"
-												onClick={ open }
-											/>
-										) }
+									<IconButton
```

Comment:
Because this component is wrapped in `<MediaUploadCheck>`, it'll only be visible for users with media upload capabilities. I think that's not intended. To test:

1. Create a user with a `contributor` role
2. Login and create a post
3. Add a cover block
4. Add some media
5. Observe that the icon button is not present on the toolbar.

Reply:
No it wasn't intended there is no need for <MediaUploadCheck> on the button. Either me misunderstanding or copy pasta :)